frontend/src/employee-frontend/components/common/PersonSearch.tsx

```diff
@@ -104,27 +102,56 @@ function PersonSearch({
     [persons] // eslint-disable-line react-hooks/exhaustive-deps
   )
 
+  const formatItemLabel = useCallback(
+    ({ firstName, lastName }: PersonDetails) =>
+      formatName(firstName, lastName, i18n),
+    [i18n]
+  )
+
+  const formatMenuItemLabel = useCallback(
+    ({
+      dateOfBirth,
+      firstName,
+      lastName,
+      streetAddress
+    }: PersonDetails): string =>
+      `${formatName(firstName, lastName, i18n)} (${dateOfBirth.format()})${
+        streetAddress ? `\n${streetAddress}` : ''
+      }`,
+    [i18n]
+  )
+
+  const filterItems = useCallback(
+    (_, items: PersonDetails[]) =>
+      searchFrom === 'vtj'
```

Comment:
Mainly thinking out loud: instead of a bunch of conditionals littered in a single component (with the risk of forgetting them in some branch), would it be better to have:

- `AbstractPersonSearch`, which takes in everything that is different between the implementations as parameters (e.g. the search method)
- `DbPersonSearch` (awful name, I know), which has no elements of its own but just passes the correct parameters to `AbstractPersonSearch`
- `VtjPersonSearch`, which does the the same but with VTJ methods/parameters

Reply:
Thank you for thinking out loud! It turned out much better, imo.

Reply:
Excellent stuff